swagger.yaml

```diff
@@ -1,83 +1,192 @@
 swagger: '2.0'
+
 info:
-  title: Web API Skeleton
-  description: Skeleton for Dropwizard Web APIs
-  version: '0.0'
-  license:
-    name: GNU Affero General Public License Version 3
-    url: http://www.gnu.org/licenses/agpl-3.0.en.html
-externalDocs:
-  description: GitHub Repository
-  url: https://github.com/osu-mist/web-api-skeleton
-host: localhost
-schemes:
-  - http
-basePath: /api/v0
-consumes:
-  - text/plain
-produces:
-  - text/plain
+  title: Music Collection API
+  version: "0.1"
+  
+#consumes: [ 'application/json' ]
+produces: [ 'application/json' ]
+
 paths:
-  /:
+  '/shelf':
+    post:
+      description: |
+        Create a shelf.
+      parameters:
+        - name: name
+          in: formData
+          type: string
+          required: false
+          description: Name of the shelf
+      responses:
+        200:
+          description: Successful response
+    
+  '/shelf/{id}':
     get:
-      summary: Get information
-      description: Build and runtime information about the application.
+      description: |
+        GET returns details about a shelf.
+      parameters:
+        - name: id
+          in: path
+          type: number
+          required: true
+          description: ID of the shelf
       responses:
         200:
-          description: Object containing information
-          schema:
-            $ref: '#/definitions/Info'
-  /sample:
+          description: Successful response
+          schema: {$ref: '#/definitions/shelf'}
+         
+    post:
+      description: |
+        POST adds an album to the shelf.
```

Comment:
If you want to add an album to a shelf the request would be: POST /shelf/{id}/albums .